In [ ]:
import numpy as np

data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

n = len(data)
mean = np.mean(data)
std_sample = np.std(data, ddof=1)

mean, std_sample

(np.float64(1.2386666666666666), np.float64(0.19316412956959936))

In [ ]:
import scipy.stats as stats

t = stats.t.ppf(0.995, df=n-1)

margin_t = t * (std_sample / np.sqrt(n))

lower_t = mean - margin_t
upper_t = mean + margin_t

lower_t, upper_t

(np.float64(1.0901973384384906), np.float64(1.3871359948948425))

In [ ]:
sigma = 0.2
z = stats.norm.ppf(0.995)

margin_z = z * (sigma / np.sqrt(n))

lower_z = mean - margin_z
upper_z = mean + margin_z

lower_z, upper_z

(np.float64(1.1056514133957607), np.float64(1.3716819199375725))

# **Final Summary**



*   Using sample standard deviation (t-distribution), the 99% confidence interval is:
(1.09 , 1.39)
*   Using known population standard deviation (z-distribution), the 99% confidence interval is:
(1.11 , 1.37)
* The t-interval is slightly wider due to uncertainty in estimating population variance from a small sample (n=15).

